### PR TITLE
Feature/gal/30 organize componant

### PR DESCRIPTION
**Pull request by galdd**
Final commit, We created different component for the ExpenseListSection, 

new components:
ExpenseListSection,
ExpenseListToolbar
ExpenseList,ExpenseListHeader
Expenses-header
Expense-item

Hook:
useExpenseLIst

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,19 +7,24 @@
...
+    "preview": "vite preview",
+    "plop": "plop"
   },
   "dependencies": {
     "@auth0/auth0-react": "^2.2.4",
     "@fortawesome/react-fontawesome": "github:fortawesome/react-fontawesome",
     "@tanstack/react-query": "^5.17.9",
     "antd": "^5.15.0",
     "date-fns": "^3.2.0",
+    "kebab-case": "^1.0.2",
```
Comment (ofir-zeitoun) [inline package.json:19]: what is it for? dev-dependency?

File: package.json [hunk 2]
```diff
@@ -7,19 +7,24 @@
...
+    "lodash": "^4.17.21",
```
Comment (ofir-zeitoun) [inline package.json:20]: needed?

File: src/@types/expense-list-prop.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { Expense } from "./expense";
+
+export type ListProps = {
```
Comment (ofir-zeitoun) [inline src/@types/expense-list-prop.ts:3]: Naming

File: src/App.css [hunk 4]
```diff
@@ -20,4 +23,9 @@ body {
   padding: 1.5em;
   will-change: filter;
   transition: filter 300ms;
+}
+
+.transparent {
+  background: transparent !important;
```
Comment (ofir-zeitoun) [inline src/App.css:29]: display: hidden

File: src/App.tsx [hunk 5]
```diff
@@ -9,13 +9,11 @@ function App() {
   return (
     <QueryWrapper>
       <NavBar />
+      <br />
```
Comment (ofir-zeitoun) [inline src/App.tsx:12]: css

File: src/components/connection-health/connection-health.css [hunk 6]
```diff
@@ -1,7 +1,7 @@
-
-.connection-wrapper{
-    padding-top: 70px;
+.connection-wrapper {
+    padding-right: 70px;
```
Comment (ofir-zeitoun) [inline src/components/connection-health/connection-health.css:2]: use rem instead

File: src/components/dashboard/dashboard.css [hunk 7]
```diff
@@ -0,0 +1,9 @@
+.dashboard {
+    display: flex;
+    justify-content: space-between;
+}
+
+.dashboard-left,
+.dashboard-right {
+    width: 48%;
```
Comment (ofir-zeitoun) [inline src/components/dashboard/dashboard.css:8]: rem

File: src/components/dashboard/dashboard.tsx [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import { ExpenseListSection } from "../expense-list-section";
+import { OverviewSection } from "../overview-section";
+import "./dashboard.css";
+
+export const Dashboard = () => {
+  return (
+    <div className="dashboard">
+      <div className="dashboard-left">
```
Comment (ofir-zeitoun) [inline src/components/dashboard/dashboard.tsx:8]: left -> overview
right -> section

File: src/components/expense-list-section/expense-list-section.tsx [hunk 9]
```diff
@@ -0,0 +1,28 @@
+import { useState } from "react";
+import ExpenseListToolbar from "./expense-list-toolbar/expense-list-toolbar";
+import { ExpenseList } from "./expense-list";
+import { useExpenseLists } from "./useExpenseLists";
+import "./expense-list-section.css";
+
+export const ExpenseListSection = () => {
+  const [sortOrder, setSortOrder] = useState<"asc" | "desc">("asc");
```
Comment (ofir-zeitoun) [inline src/components/expense-list-section/expense-list-section.tsx:8]: export to type

File: src/components/expense-list-section/expense-list-toolbar/expense-list-toolbar.tsx [hunk 10]
```diff
@@ -0,0 +1,53 @@
+import React, { useState } from "react";
+import "./expense-list-toolbar.css";
+
+interface TabProps {
+  label: string;
+  onClick: () => void;
+  isActive: boolean;
+}
+
+const Tab: React.FC<TabProps> = ({ label, onClick, isActive }) => (
```
Comment (ofir-zeitoun) [inline src/components/expense-list-section/expense-list-toolbar/expense-list-toolbar.tsx:10]: why FC?

File: src/components/expense-list-section/expense-list-toolbar/expense-list-toolbar.tsx [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+
+  const handleTabClick = (label: string) => {
+    setActiveTab(label);
+    console.log(`Active tab is now: ${label}`);
+  };
+
+  return (
+    <header className="expense-list-toolbar">
+      <nav className="tabs">
+        {["Personal", "Groups"].map((tabLabel, index) => (
```
Comment (ofir-zeitoun) [inline src/components/expense-list-section/expense-list-toolbar/expense-list-toolbar.tsx:31]: const

File: src/components/expense-list-section/expense-list-toolbar/expense-list-toolbar.tsx [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+          <Tab
+            key={index}
```
Comment (ofir-zeitoun) [inline src/components/expense-list-section/expense-list-toolbar/expense-list-toolbar.tsx:33]: [key index anti pattern](https://medium.com/geekculture/reactjs-why-index-as-a-key-is-an-anti-pattern-4b9dc6ef0067)

File: src/components/expense-list-section/expense-list-toolbar/expense-list-toolbar.tsx [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+          />
+        ))}
+      </nav>
+
+      <div className="toolbar-action">
+        <button className="toolbar-sort" onClick={toggleSortOrder}>
+          Sort
+        </button>
+        <div className="add-list-button">
+          <span>+</span> Add List
```
Comment (ofir-zeitoun) [inline src/components/expense-list-section/expense-list-toolbar/expense-list-toolbar.tsx:46]: icon?

File: src/components/expense-list-section/expense-list/expense-item/expense-item.tsx [hunk 14]
```diff
@@ -0,0 +1,53 @@
+import { EditOutlined, UserOutlined } from "@ant-design/icons";
+import "./expense-item.css";
```
Comment (ofir-zeitoun) [inline src/components/expense-list-section/expense-list/expense-item/expense-item.tsx:2]: import order

File: src/components/expense-list-section/expense-list/expense-item/expense-item.tsx [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+import { Expense } from "../../../../@types/expense";
+
+export const ExpenseItem = ({
+  item,
+  className,
+}: {
+  item: Expense;
+  className?: string;
+}) => {
+  const formatDate = (dateString: string) => {
```
Comment (ofir-zeitoun) [inline src/components/expense-list-section/expense-list/expense-item/expense-item.tsx:12]: move to utilities

File: src/components/expense-list-section/expense-list/expense-item/expense-item.tsx [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+        <div className="user-icon-wrapper">
+          <UserOutlined className="user-icon" />
+        </div>
+      )}
+      <div className="expense-details">
+        <div className="expense-title">{item.name}</div>
+      </div>
+      <div className="expense-price">${item.price.toFixed(2)}</div>
+      <div className="expense-timestamp">
+        <div className="expense-date">{formatDate(item.createdAt)}</div>
```
Comment (ofir-zeitoun) [inline src/components/expense-list-section/expense-list/expense-item/expense-item.tsx:47]: make sure createdAt is Date

File: src/components/expense-list-section/expense-list/expense-item/expense-item.tsx [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+    const date = new Date(dateString);
+    return date.toLocaleTimeString("en-GB", {
+      hour: "2-digit",
+      minute: "2-digit",
+    });
+  };
+
+  return (
+    <div className={`expense-item ${className}`}>
+      {item.creatorImageUrl ? (
```
Comment (ofir-zeitoun) [inline src/components/expense-list-section/expense-list/expense-item/expense-item.tsx:31]: create shared component

File: src/components/expense-list-section/expense-list/expense-item/expense-item.tsx [hunk 18]
```diff
@@ -0,0 +1,53 @@
+import { EditOutlined, UserOutlined } from "@ant-design/icons";
+import "./expense-item.css";
+import { Expense } from "../../../../@types/expense";
+
+export const ExpenseItem = ({
+  item,
```
Comment (ofir-zeitoun) [inline src/components/expense-list-section/expense-list/expense-item/expense-item.tsx:6]: expense

File: src/components/expense-list-section/expense-list/expense-item/expense-item.tsx [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+  className,
```
Comment (ofir-zeitoun) [inline src/components/expense-list-section/expense-list/expense-item/expense-item.tsx:7]: needed?

File: src/components/expense-list-section/expense-list/expense-item/expense-item.tsx [hunk 20]
```diff
@@ -0,0 +1,53 @@
...
+}: {
+  item: Expense;
+  className?: string;
```
Comment (ofir-zeitoun) [inline src/components/expense-list-section/expense-list/expense-item/expense-item.tsx:10]: remove. use nth css

File: src/components/expense-list-section/expense-list/expense-list-header/expense-list-header.css [hunk 21]
```diff
@@ -0,0 +1,100 @@
+.expense-list-header {
+    display: flex;
+    gap: 20px;
+    justify-content: space-between;
+    align-items: center;
+}
+
+@media (max-width: 991px) {
```
Comment (ofir-zeitoun) [inline src/components/expense-list-section/expense-list/expense-list-header/expense-list-header.css:8]: needed?

File: src/components/expense-list-section/expense-list/expense-list-header/expense-list-header.css [hunk 22]
```diff
@@ -0,0 +1,100 @@
...
+        max-width: 100%;
+        flex-wrap: wrap;
+    }
+}
+
+.expense-list-header {
+
+    .list-name {
+        color: #191919;
+        font: 700 18px Nunito, sans-serif;
```
Comment (ofir-zeitoun) [inline src/components/expense-list-section/expense-list/expense-list-header/expense-list-header.css:19]: font?

File: src/components/expense-list-section/expense-list/expense-list-header/expense-list-header.css [hunk 23]
```diff
@@ -0,0 +1,100 @@
...
+    .expense-list-header {
+        max-width: 100%;
+        flex-wrap: wrap;
+    }
+}
+
+.expense-list-header {
+
+    .list-name {
+        color: #191919;
```
Comment (ofir-zeitoun) [inline src/components/expense-list-section/expense-list/expense-list-header/expense-list-header.css:18]: css variable

File: src/components/expense-list-section/expense-list/expense-list-header/expense-list-header.tsx [hunk 24]
```diff
@@ -0,0 +1,57 @@
...
+      style: { color: "red" },
+      onClick: () => console.log("test logout"),
+    },
+  ];
+
+  return (
+    <div className="expense-list-header">
+      <h2 className="list-name">{listName || "List"}</h2>
+      <div className="list-header-actions">
+        <div className="total-amount">{expenseTotal + "$" || "350.00$"}</div>
```
Comment (ofir-zeitoun) [inline src/components/expense-list-section/expense-list/expense-list-header/expense-list-header.tsx:41]: 350? use 0$

File: src/components/expense-list-section/expense-list/expense-list-header/expense-list-header.tsx [hunk 25]
```diff
@@ -0,0 +1,57 @@
...
+    {
+      key: "Permissions",
+      icon: <SettingOutlined />,
+      label: "Permissions ",
+    },
+    {
+      key: "delete",
+      icon: <DeleteFilled />,
+      label: "Remove List",
+      style: { color: "red" },
```
Comment (ofir-zeitoun) [inline src/components/expense-list-section/expense-list/expense-list-header/expense-list-header.tsx:32]: no hardcoded style

File: src/components/expense-list-section/expense-list/expense-list.tsx [hunk 26]
```diff
@@ -0,0 +1,66 @@
...
+      >
+        Sort
+      </Button> */}
+      <ExpensesHeader />
+      <div className="sperated-expenses-list" />
+      {list.expenses?.map((item, index) => (
+        <ExpenseItem
+          key={item._id}
+          item={item}
+          className={index % 2 === 0 ? "expense-item even" : "expense-item odd"}
```
Comment (ofir-zeitoun) [inline src/components/expense-list-section/expense-list/expense-list.tsx:51]: css nth child

File: src/components/expense-list-section/expense-list/expenses-header/expenses-header.tsx [hunk 27]
```diff
@@ -0,0 +1,63 @@
+import { useState } from "react";
+import { DownOutlined, UpOutlined } from "@ant-design/icons";
+import "./expenses-header.css";
+
+type SortDirection = "asc" | "desc";
```
Comment (ofir-zeitoun) [inline src/components/expense-list-section/expense-list/expenses-header/expenses-header.tsx:5]: move to types

File: src/components/expense-list-section/expense-list/expenses-header/expenses-header.tsx [hunk 28]
```diff
@@ -0,0 +1,63 @@
...
+type SortField = "participant" | "expense" | "price" | "date";
```
Comment (ofir-zeitoun) [inline src/components/expense-list-section/expense-list/expenses-header/expenses-header.tsx:6]: const sortNames = ["participant" , "expense" , "price" , "date"] as const;
type Sort = typeof sortNames[number]

File: src/components/expense-list-section/expense-list/useExpenses.ts [hunk 29]
```diff
@@ -0,0 +1,59 @@
...
+import { Expense } from "../../../@types/expense";
+import { apiFetch } from "../../../api";
+import { useEffect, useState } from "react";
+import { useAuth0 } from "@auth0/auth0-react";
+
+async function fetchExpense(
+  token: string,
+  offset: number,
+  limit: number,
+  sortOrder: "asc" | "desc"
```
Comment (ofir-zeitoun) [inline src/components/expense-list-section/expense-list/useExpenses.ts:11]: type

File: src/components/nav-bar/nav-bar.tsx [hunk 30]
```diff
@@ -7,6 +7,7 @@ import {
 } from "@ant-design/icons";
 import { useAuth0 } from "@auth0/auth0-react";
 import "./nav-bar.css";
+import { ConnectionHealth } from "../connection-health";
```
Comment (ofir-zeitoun) [inline src/components/nav-bar/nav-bar.tsx:10]: order

File: src/components/overview-section/overview-section.tsx [hunk 31]
```diff
@@ -0,0 +1,23 @@
+import { useState } from "react";
+import "./overview-section.css";
+
+export const OverviewSection = () => {
+  const [usersCount] = useState<number>(150);
```
Comment (ofir-zeitoun) [inline src/components/overview-section/overview-section.tsx:5]: demo

File: src/components/connection-health/connection-health.css (hunk 6 above)
Comment (galdd) [reply src/components/connection-health/connection-health.css:2]: In other branch I will change any place with px to rem

File: src/components/expense-list-section/expense-list-toolbar/expense-list-toolbar.tsx (hunk 11 above)
Comment (galdd) [reply src/components/expense-list-section/expense-list-toolbar/expense-list-toolbar.tsx:31]: I dont understand the task,

File: src/components/expense-list-section/expense-list/expense-item/expense-item.tsx (hunk 19 above)
Comment (galdd) [reply src/components/expense-list-section/expense-list/expense-item/expense-item.tsx:7]: In other branch I will fix it
https://github.com/ofir-zeitoun/expenses-client/issues/33

File: src/components/expense-list-section/expense-list/expense-item/expense-item.tsx (hunk 20 above)
Comment (galdd) [reply src/components/expense-list-section/expense-list/expense-item/expense-item.tsx:10]: In other branch I will fix it
https://github.com/ofir-zeitoun/expenses-client/issues/33

File: src/components/expense-list-section/expense-list/expense-list-header/expense-list-header.css (hunk 23 above)
Comment (galdd) [reply src/components/expense-list-section/expense-list/expense-list-header/expense-list-header.css:18]: in other branch I will fix it 
https://github.com/ofir-zeitoun/expenses-client/issues/33

File: src/components/expense-list-section/expense-list/expense-list.tsx (hunk 26 above)
Comment (galdd) [reply src/components/expense-list-section/expense-list/expense-list.tsx:51]: in other branch I will fix it
https://github.com/ofir-zeitoun/expenses-client/issues/33

File: src/components/overview-section/overview-section.tsx (hunk 31 above)
Comment (galdd) [reply src/components/overview-section/overview-section.tsx:5]: other branch

**Review by ofir-zeitoun — APPROVED** [review]

**Pull request merged**